crypto/ocsp/ocsp_cl.c

```diff
@@ -214,6 +215,26 @@ int OCSP_resp_get0_id(const OCSP_BASICRESP *bs,
     return 1;
 }
 
+int OCSP_resp_get1_id(const OCSP_BASICRESP *bs,
+                      ASN1_OCTET_STRING **pid,
+                      X509_NAME **pname)
+{
+    const OCSP_RESPID *rid = &bs->tbsResponseData.responderId;
+
+    if (rid->type == V_OCSP_RESPID_NAME) {
+        *pname = X509_NAME_dup(rid->value.byName);
+        *pid = NULL;
+    } else if (rid->type == V_OCSP_RESPID_KEY) {
+        *pid = ASN1_OCTET_STRING_dup(rid->value.byKey);
+        *pname = NULL;
+    } else {
+        return 0;
+    }
+    if (!pname && !pid)
+        return -1;
```

Comment:
The coding style says that "usually" we use 1 for success and 0 for failure, and that "sometimes" an additional value is used "-1: something bad (e.g., internal error or memory allocation failure)". Perhaps the wording could do with being clarified. It is intended to convey that we use the additional value on occasions where it is necessary to distinguish a "normal" failure and a "really bad" failure. This isn't one of those occasions. It is enough to know that it failed.